src/Tools/Console/Command/GenerateCommand.php

```diff
@@ -57,9 +60,20 @@ protected function execute(InputInterface $input, OutputInterface $output): int
         }
 
         $dirs = $configuration->getMigrationDirectories();
-        if ($namespace === null) {
+        if ($namespace === null && count($dirs) === 1) {
             $namespace = key($dirs);
-        } elseif (! isset($dirs[$namespace])) {
+        } elseif ($namespace === null && count($dirs) > 1) {
+            $helper    = $this->getHelper('question');
+            $question  = new ChoiceQuestion(
+                'Please choose a namespace (defaults to the first one)',
+                array_keys($dirs),
+                0,
+            );
+            $namespace = $helper->ask($input, $output, $question);
+            $this->io->text(sprintf('You have selected the "%s" namespace', $namespace));
+        }
```

Comment:
This is quite repetitive… would it make sense to introduce a `DoctrineCommand::getNamespace()` method so as to enforce a consistent experience?

Reply:
I agree with you. I had this thought :)